InstanceExport/PowerShell/InstanceExport.ps1

```diff
@@ -178,71 +186,87 @@ This function exports an given instancce and save it to your disk in the specifi
                     CreateDirectoryIfDoesNotExist -directoryPath $directoryPath -currentDate $currentDate
                 }
-                catch{
+                catch {
                     $msg = "Error trying to create $directoryPath directory"
                     $displayMsg = "An error occurred when trying to create a new directory"
                     Log -msg $msg -displayMsg $displayMsg -logLevel "error" -currentDate $currentDate
                 }
                 $fileName = $entry.FileName.Split([IO.Path]::GetInvalidFileNameChars()) -join ''
                 $filePath = "$directoryPath/$fileName"
                 $fileAlreadyExists = Test-Path -Path $filePath -PathType Leaf
-                if (!$fileAlreadyExists -or $overwrite)
-                {
+                if (!$fileAlreadyExists -or $overwrite) {
                     $entryExportParameters = @{
                         Method      = "GET"
                         Uri         = "https://$($manifestJson.SubDomain).$($manifestJson.HostName)$($entry.Url)"
                         Headers     = $header
                         ContentType = "application/json"  
                     }
                         
-                    try{
-                        if (!$entry.fileName.Contains(".json")){
+                    try {
+                        if (!$entry.fileName.Contains(".json")) {
                             Invoke-RestMethod @entryExportParameters -OutFile $filePath | Out-Null
                         }
-                        else{
+                        else {
                             $GetEntriesResponse = Invoke-RestMethod @entryExportParameters
                             $GetEntriesResponse | ConvertTo-Json -Depth 100 | Out-File -FilePath $filePath
                         }
                     }
-                    catch{
+                    catch {
                         $message = "Error occurred when extracting data from $($entry.Url)"
                         Log -msg "An error occurred" -displayMsg $message -logLevel "error" -currentDate $currentDate
                         Log -msg $message -logLevel "error" -currentDate $currentDate
                         Log -msg "StatusCode: $($_.Exception.Response.StatusCode.value__)" -logLevel "error" -currentDate $currentDate
                         Log -msg "Url: $($entry.Url)" -logLevel "error" -currentDate $currentDate
                         Log -msg "StatusDescription: $($_.Exception.Response.StatusDescription)" -logLevel "error" -currentDate $currentDate
+                        Log -msg "Exception: $($_.Exception)" -logLevel "error" -currentDate $currentDate
+                        Write-Verbose $_.Exception
+                        Write-Verbose $_.Exception.Response
+                        if ($_.Exception.Response.StatusCode -eq 503) {
+                            $message = "Server is offline. Waiting it to get back."
+                            Write-Verbose $message
```

Comment:
Does it make sense for this message to be shown to the user so that they can know what's happening?

```suggestion
                            $message = "Server is offline. Waiting for it to become available..."
                            Write-ColorOutput red $message
```

Reply:
For my two cents, I'd prefer to be a little less transparent here. Something like "Lost connection to server. Waiting to reestablish connection."

Reply:
I've updated with your considerations. It should be good for another review